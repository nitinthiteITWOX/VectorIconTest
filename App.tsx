import { StatusBar } from 'expo-status-bar';
import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { Avatar, Button, TextInput } from 'react-native-paper';

const App: React.FC = () => {
  return (
    <View style={styles.container}>
            <Avatar.Icon style={{ backgroundColor: "#00C33F" }} color={"#FFFFFF"} icon="check" />
            <View style={{ width: "100%" }}>
            <TextInput
                        mode={"flat"}
                        label="E-mail"
                        left={<TextInput.Icon name="email-outline" color="#FF00FF" />}
                        keyboardType={"email-address"}
                        returnKeyType={"done"}
                        blurOnSubmit={true}
                    />
            </View>
        </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default App;
