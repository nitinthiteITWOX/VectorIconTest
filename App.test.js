import React from 'react';
import { App } from './App';
import { create } from 'react-test-renderer';

describe("<App /> snapshot verification", () => {

    /**
     * Method called to verify snapshot for mentioned component
     */
     test(`should work succesfully`, async () => {
        const tree = create(<App />).toJSON();
        await expect(tree).toMatchSnapshot();
});
});