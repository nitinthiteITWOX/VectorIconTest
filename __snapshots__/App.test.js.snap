// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> snapshot verification should work succesfully 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    color="#FFFFFF"
    style={
      Array [
        Object {
          "backgroundColor": "#00C33F",
          "borderRadius": 32,
          "height": 64,
          "width": 64,
        },
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        Object {},
      ]
    }
  >
    <Text />
  </View>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(231, 231, 231)",
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
          },
          Object {},
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.26)",
            "bottom": 0,
            "height": 2,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "transform": Array [
              Object {
                "scaleY": 0.5,
              },
            ],
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "paddingBottom": 0,
              "paddingTop": 0,
            },
            Object {
              "minHeight": 64,
            },
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <Text
            numberOfLines={1}
            onLayout={[Function]}
            style={
              Object {
                "color": "#6200ee",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 0,
                "paddingLeft": 44,
                "paddingRight": 12,
                "position": "absolute",
                "textAlign": "left",
                "top": 34,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            E-mail
          </Text>
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 0,
                "paddingLeft": 44,
                "paddingRight": 12,
                "position": "absolute",
                "textAlign": "left",
                "top": 34,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            E-mail
          </Text>
        </View>
        <TextInput
          allowFontScaling={true}
          blurOnSubmit={true}
          editable={true}
          keyboardType="email-address"
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder=""
          placeholderTextColor="rgba(0, 0, 0, 0.54)"
          rejectResponderTermination={true}
          returnKeyType="done"
          selectionColor="#6200ee"
          style={
            Array [
              Object {
                "flexGrow": 1,
                "margin": 0,
                "zIndex": 1,
              },
              Object {
                "paddingLeft": 44,
                "paddingRight": 12,
              },
              Object {
                "height": 64,
              },
              Object {
                "paddingBottom": 4,
                "paddingTop": 24,
              },
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              Object {
                "marginLeft": 36,
                "paddingLeft": 8,
              },
            ]
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 24,
              "justifyContent": "center",
              "position": "absolute",
              "width": 24,
            },
            Object {
              "left": 12,
              "top": 20,
            },
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": undefined,
            }
          }
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 18,
                  "height": 36,
                  "width": 36,
                },
                undefined,
                Object {
                  "margin": 0,
                },
              ],
            ]
          }
        >
          <View>
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
